src/coreml.cs

```diff
@@ -584,6 +754,233 @@ interface MLModelConfiguration : NSCopying, NSSecureCoding {
+
+		[Static]
+		[Export ("miniBatchSize")]
+		MLParameterKey MiniBatchSize { get; }
+
+		[Static]
+		[Export ("beta1")]
+		MLParameterKey Beta1 { get; }
+
+		[Static]
+		[Export ("beta2")]
+		MLParameterKey Beta2 { get; }
+
+		[Static]
+		[Export ("eps")]
+		MLParameterKey Eps { get; }
+
+		[Static]
+		[Export ("epochs")]
+		MLParameterKey Epochs { get; }
+
+		[Static]
+		[Export ("shuffle")]
+		MLParameterKey Shuffle { get; }
+
+		[Static]
+		[Export ("seed")]
+		MLParameterKey Seed { get; }
+
+		[Static]
+		[Export ("numberOfNeighbors")]
+		MLParameterKey NumberOfNeighbors { get; }
+
+		// From MLParameterKey (MLLinkedModelParameters)
+
+		[Static]
+		[Export ("linkedModelFileName")]
+		MLParameterKey LinkedModelFileName { get; }
+
+		[Static]
+		[Export ("linkedModelSearchPath")]
+		MLParameterKey LinkedModelSearchPath { get; }
+
+		// From MLParameterKey (MLNeuralNetworkParameters)
+
+		[Static]
+		[Export ("weights")]
+		MLParameterKey Weights { get; }
+
+		[Static]
+		[Export ("biases")]
+		MLParameterKey Biases { get; }
+
+		// From MLParameterKey (MLScopedParameters)
+
+		[Export ("scopedTo:")]
+		MLParameterKey GetScopedParameter (string scope);
+
+	}
+
+	[Watch (6,0), TV (13,0), Mac (10,15), iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface MLTask {
+
+		[Export ("taskIdentifier")]
+		string TaskIdentifier { get; }
+
+		[Export ("state", ArgumentSemantic.Assign)]
+		MLTaskState State { get; }
+
+		[NullAllowed, Export ("error", ArgumentSemantic.Copy)]
+		NSError Error { get; }
+
+		[Export ("resume")]
+		void Resume ();
+
+		[Export ("cancel")]
+		void Cancel ();
+	}
+
+	[Watch (6,0), TV (13,0), Mac (10,15), iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface MLUpdateContext {
+
+		[Export ("task")]
+		MLUpdateTask Task { get; }
+
+		[Export ("model")]
+		IMLWritable Model { get; }
+
+		[Export ("event")]
+		MLUpdateProgressEvent Event { get; }
+
+		[Export ("metrics")]
+		NSDictionary<MLMetricKey, NSObject> Metrics { get; }
+
+		[Export ("parameters")]
+		NSDictionary<MLParameterKey, NSObject> Parameters { get; }
+	}
+
+	[Watch (6,0), TV (13,0), Mac (10,15), iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface MLUpdateProgressHandlers {
+
+		[Export ("initForEvents:progressHandler:completionHandler:")]
+		IntPtr Constructor (MLUpdateProgressEvent interestedEvents, [NullAllowed] Action<MLUpdateContext> progressHandler, Action<MLUpdateContext> completionHandler);
+	}
+
+	[Watch (6,0), TV (13,0), Mac (10,15), iOS (13,0)]
+	[BaseType (typeof (MLTask))]
+	[DisableDefaultCtor]
+	interface MLUpdateTask {
+
+		[Static]
+		[Export ("updateTaskForModelAtURL:trainingData:configuration:completionHandler:error:")]
+		[return: NullAllowed]
+		MLUpdateTask Create (NSUrl modelUrl, IMLBatchProvider trainingData, [NullAllowed] MLModelConfiguration configuration, Action<MLUpdateContext> completionHandler, [NullAllowed] out NSError error);
```

Comment:
`[Async]` ?
I think our missing-async test will not spot such case since it's not the last parameter !?!
In fact I'm not sure if the generator would be happy...

Reply:
I do not think `[Async]` is needed here since this creates a Task object but its completion handler is not called when this task object is finished being created but when the task is completed